src/commands/heroku.rs

```diff
@@ -390,6 +398,177 @@ pub fn restart_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandR
     Ok(())
 }
 
+#[command]
+#[num_args(3)]
+pub fn deploy_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
+    let app_name = args
+        .single::<String>()
+        .expect("You must include an app name");
+
+    let app_version = args
+        .single::<String>()
+        .expect("You must include an app version");
+
+    let git_ref = args
+        .single::<String>()
+        .expect("You must include a git ref to deploy");
+
+    let github_client = Github::new(bot_config(ctx).github_token.to_string());
+
+    if github_client.is_err() {
+        msg.reply(
+            &ctx,
+            format!(
+                "An error occured when trying to contact GitHub with your GitHub token:\n{:?}",
+                github_client.err()
+            ),
+        )?;
+
+        return Ok(());
+    }
+
+    let github_response = github_client
+        .unwrap()
+        .get()
+        .custom_endpoint(&github_ref_endpoint(bot_config(ctx), git_ref))
+        .execute::<Value>();
+
+    if github_response.is_err() {
+        msg.reply(
+            &ctx,
+            format!(
+                "An error occured when trying to get commit info for {}/{}:\n{:?}",
+                bot_config(ctx).github_org,
+                bot_config(ctx).github_repo,
+                github_response.as_ref().err()
+            ),
+        )?;
+    }
+
+    let git_sha_value = &github_response.unwrap().2.unwrap()["sha"];
+    let git_sha = git_sha_value.as_str().unwrap();
+
+    let build_create_response = heroku_client(ctx).request(&builds::BuildCreate {
+        app_id: app_name.clone(),
+        params: builds::BuildCreateParams {
+            buildpacks: None,
+            source_blob: builds::SourceBlobParam {
+                checksum: None,
+                url: source_url(&ctx, &git_sha.to_string()),
+                version: Some(app_version.clone()),
```

Comment:
As far as I understand we don't really have a concept of "crates.io versions", I'd just use the SHA here.